### PR TITLE
test_externals added for near-vm-runner

### PR DESCRIPTION
**Pull request by fckt**
Rebased to staging

### CODE REVIEW

File: runtime/near-vm-runner/tests/test-contract-rs/src/lib.rs [hunk 1]
```diff
@@ -87,116 +87,140 @@ extern "C" {
...
 #[no_mangle]
-pub fn write_key_value() {
-    unsafe {
-        input(0);
-        if register_len(0) != 2 * size_of::<u64>() as u64 {
-            panic()
-        }
-        let data = [0u8; 2 * size_of::<u64>()];
-        read_register(0, data.as_ptr() as u64);
+pub unsafe fn ext_predecessor_account_id() {
```
Comment (nearmax) [inline runtime/near-vm-runner/tests/test-contract-rs/src/lib.rs]: Could probably write a macro to avoid repetitions.

**Comment by nearmax** [comment]
This tests only 4 external methods. Could you add coverage for other external methods? Also, we should test that `near-vm-logic` follows the spec in https://github.com/nearprotocol/NEPs/pull/9 including all corner cases.

**Comment by nearmax** [comment]
Fixes: #1292
